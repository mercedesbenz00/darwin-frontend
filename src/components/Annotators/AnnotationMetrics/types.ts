import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
