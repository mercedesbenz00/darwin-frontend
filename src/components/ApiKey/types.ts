export type AbilityGroup =
  'manage_comments' |
  'manage_tasks' |
  'manage_datasets' |
  'manage_team' |
  'manage_team_members' |
  'manage_models'

export type Group = { id: AbilityGroup; name: string }
export type GroupOption = Group & { selected: boolean }

export enum SupportedLanguages {
  shell = 'shell',
  python = 'python',
  javascript = 'javascript',
  elixir = 'elixir',
  cli = 'cli'
}

export type AccessCodeExample = { language: SupportedLanguages, code: string }
