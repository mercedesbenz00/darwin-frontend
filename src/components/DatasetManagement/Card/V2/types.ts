import {
  DatasetItemPayload,
  V2DatasetItemPayload,
  V2DatasetFolderPayload
} from '@/store/types'

export type DatasetFolderCardProps = {
  data: V2DatasetFolderPayload
  readonly: boolean
  urlPrefix: string | null
}

export type DatasetItemCardProps = {
  data: DatasetItemPayload,
  dataV2: V2DatasetItemPayload
}
