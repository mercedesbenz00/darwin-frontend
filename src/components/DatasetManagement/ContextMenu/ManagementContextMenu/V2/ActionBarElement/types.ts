export enum ActionBarType {
  DEFAULT = 'default',
  SELECT = 'select',
  DANGER = 'danger'
}

export type ActionBarElement = {
  active?: boolean
  disabled?: string
}

export enum DangerMode {
  ARCHIVE = 'archive',
  DELETE = 'delete'
}
