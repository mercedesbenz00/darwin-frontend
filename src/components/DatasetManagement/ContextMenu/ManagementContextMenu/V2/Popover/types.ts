export enum PopoverContextType {
  ASSIGN = 'assign',
  STAGE = 'stage',
  FOLDER = 'folder',
  PRIORITY = 'priority'
}
