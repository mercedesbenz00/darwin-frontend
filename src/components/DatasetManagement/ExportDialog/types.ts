import { DropdownOption } from '@/components/Common/Dropdown/types'

export type ExportFormatPillType = {
  name: string;
  type: 'feather-light' | 'pink' | 'yellow';
}

export type ExportFormatOptionType = DropdownOption & {
  id: string
  label: string
  pills?: ExportFormatPillType[]
}
