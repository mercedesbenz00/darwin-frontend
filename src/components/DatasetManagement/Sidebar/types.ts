import { DatasetItemStatus } from '@/store/types'

export type ClassFilterChange = {
  positiveOptions: number[]
  negativeOptions: number[]
}

export type StatusFilterChange = {
  commented: boolean
  positiveOptions: DatasetItemStatus[]
  negativeOptions: DatasetItemStatus[]
}

export type GenericFilterChange = {
  positiveAssignees: number[]
  negativeAssignees: number[]
  positiveFilenames: string[]
  negativeFilenames: string[]
  positivePaths: string[]
  negativePaths: string[]
}
