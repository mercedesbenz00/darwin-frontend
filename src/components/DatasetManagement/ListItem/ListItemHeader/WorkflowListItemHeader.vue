<template>
  <div class="list-item-header">
    <list-item-header-item
      v-for="({ id, title, sortable }) of columns"
      :id="id"
      :key="id"
      :class="`list-item-header__${id}`"
      :title="title"
      :sortable="sortable"
      :active="id === activeSortBy"
      :sort-direction="sortable && (id === activeSortBy ? activeSortDirection : 'asc')"
      @click="onColumnClick"
    />
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { Getter } from 'vuex-class'

import { DatasetPayload } from '@/store/types'

import ListItemHeaderItem from './ListItemHeaderItem.vue'

@Component({
  name: 'list-item-header',
  components: { ListItemHeaderItem }
})
export default class ListItemHeader extends Vue {
  @Prop({ required: true })
  dataset!: DatasetPayload

  @Getter('dataTabSortBy', { namespace: 'dataset' })
  getActiveSortBy!: (dataset: DatasetPayload) => string

  @Getter('dataTabSortDirection', { namespace: 'dataset' })
  getActiveSortDirection!: (dataset: DatasetPayload) => 'asc' | 'desc'

  get activeSortBy (): string {
    return this.getActiveSortBy(this.dataset)
  }

  get activeSortDirection (): 'asc' | 'desc' {
    return this.getActiveSortDirection(this.dataset)
  }

  get columns () {
    return [
      { id: 'checkbox' },
      { id: 'thumbnail' },
      { id: 'file_name', title: 'Image Name' },
      { id: 'status', title: 'Status' },
      { id: 'inserted_at', title: 'Date Added', sortable: true },
      { id: 'modified_at', title: 'Date Modified', sortable: true },
      { id: 'labels', title: 'Tags' },
      { id: 'image_dimension', title: 'Pixels', sortable: true },
      { id: 'file_size', title: 'Size', sortable: true }
    ]
  }

  onColumnClick (params: { id: string, sortDirection: number }) {
    const { id, sortDirection } = params
    // Update the sort query in the url
    this.$router.push({
      query: {
        ...this.$route.query,
        sort: `${id}:${sortDirection}`
      }
    })
  }
}
</script>

<style lang="scss" scoped>
@import "workflow-list-item";

.list-item-header {
  @include workflow-list-item;
  padding-right: 10px;
  box-shadow: 0 3px 10px -5px rgba(145, 169, 192, .3);
  border-radius: 3px;
  margin-bottom: 0;
}

.list-item-header__file_name {
  display: block;
}
</style>
