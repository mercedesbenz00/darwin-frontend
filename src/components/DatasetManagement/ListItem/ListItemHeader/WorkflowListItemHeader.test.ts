import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'

import createTestStore from 'test/unit/createTestStore'
import { buildDatasetItemFilter, buildDatasetPayload } from 'test/unit/factories'

import WorkflowListItemHeader from '@/components/DatasetManagement/ListItem/ListItemHeader/WorkflowListItemHeader.vue'
import { DatasetPayload } from '@/store/types'

const localVue = createLocalVue()
localVue.use(Vuex)

let propsData: { dataset: DatasetPayload }
let store: ReturnType<typeof createTestStore>

beforeEach(() => {
  propsData = {
    dataset: buildDatasetPayload({ work_prioritization: 'inserted_at:desc' })
  }
  store = createTestStore()
})

it('matches snapshot', () => {
  const wrapper = shallowMount(WorkflowListItemHeader, { localVue, propsData, store })
  expect(wrapper).toMatchSnapshot()
})

it('commits new option when selected', () => {
  const mocks = {
    $route: { name: 'DatasetManagementData', params: {}, query: {} },
    $router: { push: jest.fn() }
  }
  store.commit(
    'dataset/SET_DATASET_ITEMS_FILTER',
    buildDatasetItemFilter({ sort: { priority: 'asc' } })
  )
  const wrapper = shallowMount(WorkflowListItemHeader, { localVue, mocks, propsData, store })
  jest.spyOn(store, 'commit').mockReturnValue(undefined)
  const clickEvent = { id: 'inserted_at', sortDirection: 'desc' }
  wrapper.find('.list-item-header__inserted_at').vm.$emit('click', clickEvent)
  expect(mocks.$router.push).toBeCalledWith({
    query: { sort: 'inserted_at:desc' }
  })
})
