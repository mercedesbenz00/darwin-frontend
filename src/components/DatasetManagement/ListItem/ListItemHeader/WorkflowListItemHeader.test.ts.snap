// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div class="list-item-header">
  <list-item-header-item-stub id="checkbox" title="" sortdirection="asc" class="list-item-header__checkbox"></list-item-header-item-stub>
  <list-item-header-item-stub id="thumbnail" title="" sortdirection="asc" class="list-item-header__thumbnail"></list-item-header-item-stub>
  <list-item-header-item-stub id="file_name" title="Image Name" sortdirection="asc" class="list-item-header__file_name"></list-item-header-item-stub>
  <list-item-header-item-stub id="status" title="Status" sortdirection="asc" class="list-item-header__status"></list-item-header-item-stub>
  <list-item-header-item-stub id="inserted_at" title="Date Added" sortdirection="desc" active="true" sortable="true" class="list-item-header__inserted_at"></list-item-header-item-stub>
  <list-item-header-item-stub id="modified_at" title="Date Modified" sortdirection="asc" sortable="true" class="list-item-header__modified_at"></list-item-header-item-stub>
  <list-item-header-item-stub id="labels" title="Tags" sortdirection="asc" class="list-item-header__labels"></list-item-header-item-stub>
  <list-item-header-item-stub id="image_dimension" title="Pixels" sortdirection="asc" sortable="true" class="list-item-header__image_dimension"></list-item-header-item-stub>
  <list-item-header-item-stub id="file_size" title="Size" sortdirection="asc" sortable="true" class="list-item-header__file_size"></list-item-header-item-stub>
</div>
`;
